solutions/level1/cart.rb

```diff
@@ -6,39 +6,28 @@ class Cart
   attr_accessor :total, :item_count
 
   def initialize
-    Book.clear_books
     @total = 0
     @item_count = 0
-    @json_receipt = {books: [], cart: {total: 0, item_count: 0}}
   end
 
 
-  def calculate_receipt(books, book_ids)
+  def calculate_totals(books, book_ids)
     unless books.empty?
       books.each do |book|
         if book_ids.include?(book.id)
           @total += book.price
           @item_count += 1
         end
-
-        @json_receipt[:books].append(book.to_hash)
-        @json_receipt[:cart][:total] = @total
-        @json_receipt[:cart][:item_count] = @item_count
       end
     end
   end
 
-  def export_receipt(file, book_ids)
-    data_hash = JsonReaderWriter.read_file(file)
-    data_hash['books'].each do |book_hash|
-      Book.books << Book.new(book_hash['id'], book_hash['title'], book_hash['price'])
-    end
-
-    calculate_receipt(Book.books, book_ids)
+  def generate_receipt(books)
+    books = BookSerializer.serialize_books(books)
```

Comment:
instead of having a serializer, why not simply define a `to_h` or `to_json` method on the record and have that return what is needed ?